autoload/lsp/internal/diagnostics/float.vim

```diff
@@ -16,7 +16,14 @@ function! lsp#internal#diagnostics#float#_enable() abort
     let s:enabled = 1
 
     let s:Dispose = lsp#callbag#pipe(
-        \ lsp#callbag#fromEvent('CursorMoved'),
+        \ lsp#callbag#merge(
+        \   lsp#callbag#fromEvent(['CursorMoved', 'CursorHold']),
+        \   lsp#callbag#pipe(
+        \       lsp#callbag#fromEvent(['InsertEnter']),
+        \       lsp#callbag#filter({_->!g:lsp_diagnostics_float_insert_mode_enabled}),
+        \       lsp#callbag#tap({_->s:hide_float()}),
+        \   )
+        \ ),
```

Comment:
I added it based on document_highlight.

https://github.com/prabirshrestha/vim-lsp/blob/500987604d356738068ee3bf320a82dfa9fbfc1f/autoload/lsp/internal/document_highlight.vim#L20-L25

The reason for adding `CursorHold` is to show the float diagnostic error even if the cursor isn't moved.